relay/pkg/block/push.go

```diff
@@ -0,0 +1,214 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+)
+
+// pullHeadersFromQueue waits until we have `headersBatchSize` headers from
+// the queue or until the queue fails to yield a header for
+// `headerTimeout` duration.
+func (f *Forwarder) pullHeadersFromQueue(ctx context.Context) []*btc.Header {
```

Comment:
Thoughts on naming it `pullHeaders` or `pullBtcHeaders`? The fact we take them from `f.headersQueue` is an implementation details. From the perspective of the code calling this function, we are pulling them from Bitcoin chain.

Reply:
I'm rather against it because within the forwarder we pull block either from the BTC chain or from the queue and naming it just `pullHeaders` will be ambiguous. I think it's more readable and consistent if we have on the push side:
- `pullHeadersFromQueue`
- `pushHeadersToHostChain`
and on the pull side (see https://github.com/keep-network/tbtc/pull/769):
- `pullHeadersFromBtcChain`
- `pushHeadersToQueue`

Last but not least, those methods are package-private helpers, the client code just do `block.RunForwarder` and doesn't care.